calico_kubernetes/calico_kubernetes.py

```diff
@@ -141,6 +137,9 @@ def status(self, namespace, pod_name, docker_id):
                 workload_id=self.docker_id
             )
         except KeyError:
+            # Obtain information from Docker Client and validate container state
+            # Will exit safely (0) in host NetworkMode
+            self._validate_container_state(self.docker_id)
```

Comment:
I like your solution, but we need to use my exit code handling so that the kubelet doesn't think we failed a status check.